File: api/hello.js
// Este é o formato de handler padrão do Vercel
// É muito similar a um handler do Express.js
export default function handler(request, response) {
  // Set CORS headers
  response.setHeader('Access-Control-Allow-Origin', '*'); // Allow all origins for development
  response.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS'); // Allow GET and OPTIONS methods
  response.setHeader('Access-Control-Allow-Headers', 'Content-Type'); // Allow Content-Type header

  // Handle preflight OPTIONS request
  if (request.method === 'OPTIONS') {
    return response.status(200).end();
  }

  // request.query contém os parâmetros da URL (ex: /api/hello?name=John)
  const { name } = request.query;

  // Envia uma resposta
  response.status(200).send(`Olá, ${name || 'mundo'}!`);
}

File: api/hello.py
from http.server import BaseHTTPRequestHandler
from urllib.parse import urlparse, parse_qs
from api._utils import send_cors_preflight, send_text_response

class handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        """Handle CORS preflight request"""
        send_cors_preflight(self, 'GET, OPTIONS')
    
    def do_GET(self):
        """Handle GET request"""
        # Parse query parameters
        parsed_path = urlparse(self.path)
        query_params = parse_qs(parsed_path.query)
        name = query_params.get('name', ['mundo'])[0]
        
        # Send response
        message = f"Olá, {name}!"
        send_text_response(self, message, methods='GET, OPTIONS')
